cmd/snap-confine/snap-confine-invocation.c

```diff
@@ -144,3 +145,66 @@ void sc_check_rootfs_dir(sc_invocation *inv) {
 
     die("cannot locate base snap %s", inv->base_snap_name);
 }
+
+static char* read_homedirs_from_system_params(void)
+{
+    FILE *f SC_CLEANUP(sc_cleanup_file) = NULL;
+    f = fopen("var/lib/snapd/system-params", "r");
+    if (f == NULL) {
+        return NULL;
+    }
+
+    char *line SC_CLEANUP(sc_cleanup_string) = NULL;
+    size_t line_size = 0;
+    while (getline(&line, &line_size, f) != -1) {
+        if (sc_startswith(line, "homedirs=")) {
+            return strdup(line + (sizeof("homedirs=") - 1));
+        }
+    }
+    return NULL;
+}
+
+void sc_invocation_check_homedirs(sc_invocation *inv)
+{
+    char *config_line SC_CLEANUP(sc_cleanup_string) = read_homedirs_from_system_params();
+    if (config_line == NULL) {
+        return;
+    }
+
+    /* The homedirs setting is a comma-separated list. In order to allocate the
+     * right number of strings, let's count how many commas we have.
+     */
+    int num_commas = 0;
+    for (char *c = config_line; *c != '\0'; c++) {
+        if (*c == ',') {
+            num_commas++;
+        }
+    }
+
+    /* We add *two* elements here: one is because of course the number of
+     * actual homedirs is the number of commas plus one, and the extra one is
+     * used as an end-of-array indicator. */
+    inv->homedirs = calloc(num_commas + 2, sizeof(char *));
+    if (inv->homedirs == NULL) {
+        die("cannot allocate memory for homedirs");
+    }
+
+    // strtok_r needs a pointer to keep track of where it is in the
+    // string.
+    char *buf_saveptr = NULL;
+
+    int current_index = 0;
+    char *homedir = strtok_r(config_line, ",\n", &buf_saveptr);
+    while (homedir != NULL) {
+        if (homedir[0] == '\0') {
+            /* The typical case where we'd get an empty string is
```

Comment:
(super nitpick) I would make this comment shorter, something like: `// Deal with cases like "homedirs=", "homedirs=,," etc` (it's nice but takes too much room IMHO for what it does explains :)

Reply:
Absolutely agree! It's more sensible to be shorter about this